binaryfile.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+total=0
+
+for i in $@; do
+   if [ ! -z "${i##[0-9]*}" ]; then
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/Amruta101998/DS-and-Algorithm/pull/15#issuecomment-2181197000>#b28f0c</a><span> - 06/20/2024, 05:33 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The condition '[ ! -z "${i##[0-9]*}" ]' is used to check if the input is numeric. However, this condition is not robust and can fail for certain edge cases, such as negative numbers or numbers with leading zeros.
</div>
<div id="fix">
<b>Fix:</b> Use a more robust regular expression to check if the input is numeric, which can handle negative numbers and numbers with leading zeros.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -6,7 +6,7 @@
  for i in $@; do
 -   if [ ! -z "${i##[0-9]*}" ]; then
 +   if ! [[ "$i" =~ ^-?[0-9]+$ ]]; then
         echo "Please enter numeric only"
         echo "Hello world code"
         exit 1
```
</div>
</details>
</div>
<br>


</div>